packages/block-editor/src/store/actions.js

```diff
@@ -1172,7 +1171,7 @@ export const duplicateBlocks = ( clientIds, updateSelection = true ) => ( {
 		last( castArray( clientIds ) )
 	);
 	const clonedBlocks = blocks.map( ( block ) =>
-		__experimentalCloneSanitizedBlock( block )
+		cloneBlock( block, {}, null, { retainCopyAttributes: false } )
```

Comment:
wouldn't cloning imply retaining?

same question as above with regards to being overly-specific to `copy`

Reply:
I missed updating `retainCopyAttributes` here, I'll update that when I settle on the API.

> wouldn't cloning imply retaining?

In the driving use cases of a `productId` or `widgetId`, we don't want it retained on `copy` (whether that be from the block menu or keyboard commands) or on duplication. With this API though we could consider those to be separate supports:
```
"__experimentalSupports": {
    "copy": false,
    "duplicate": false
}
```

In either case, I think it's a good point that adding the logic to `cloneBlock` is unnecessarily confusing -- we could clone the block and strip the attributes afterward, here in the duplication logic. In an earlier iteration it made a little more sense to add the logic to `cloneBlock` because we were using it in multiple places.